nat/nat.go

```diff
@@ -92,20 +92,21 @@ func (p Port) Range() (int, int, error) {
 	return ParsePortRangeToInt(p.Port())
 }
 
-// SplitProtoPort splits a port in the format of proto/port
+// SplitProtoPort splits a port(range) and protocol, formatted as "<portnum>/[<proto>]"
+// "<startport-endport>/[<proto>]". It returns an empty string for both if
+// no port(range) is provided. If a port(range) is provided, but no protocol,
+// the default ("tcp") protocol is returned.
+//
+// SplitProtoPort does not validate or normalize the returned values.
 func SplitProtoPort(rawPort string) (string, string) {
```

Comment:
Splitting `"port/proto"` and returning `proto, port` (instead of `port, proto`) is surprising. Could name the return vals? ...

```suggestion
// SplitProtoPort does not validate or normalize the returned values.
func SplitProtoPort(rawPort string) (proto, port string) {

Reply:
Ah, yes, can do; I was actually considering that, as I also was a bit surprised that this one flipped the order of the fields (was even considering to mark it deprecated as it's a bit of a silly function, but the replacement (use `strings.Cut`) would also require users to fill in the default (`tcp`), so there's still "some" value.